codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/protocols/parse/ParserUtil.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.rust.codegen.smithy.protocols.parse
+
+import software.amazon.smithy.model.shapes.BlobShape
+import software.amazon.smithy.model.shapes.Shape
+import software.amazon.smithy.model.shapes.TimestampShape
+import software.amazon.smithy.rust.codegen.rustlang.Writable
+import software.amazon.smithy.rust.codegen.rustlang.rust
+import software.amazon.smithy.rust.codegen.rustlang.writable
+import software.amazon.smithy.rust.codegen.smithy.RuntimeConfig
+import software.amazon.smithy.rust.codegen.smithy.RuntimeType
+import software.amazon.smithy.rust.codegen.smithy.RustSymbolProvider
+import software.amazon.smithy.rust.codegen.smithy.rustType
+
+/*
+ * Utility class used to force casting a non primitive type into one overriden by a new symbol provider,
+ * by explicitly calling `from()`.
+ *
+ * For example we use this in the server Python implementation, where we override types like [Blob] and [DateTime]
+ * with wrappers compatibile with Python, without touching the original implementation coming from `aws-smithy-types`.
+ */
+class ParserUtil(private val symbolProvider: RustSymbolProvider, private val runtimeConfig: RuntimeConfig) {
+    fun convertViaFrom(shape: Shape): Writable =
```

Comment:
How this is written now assumes the conversion will always be a `.something()` suffix, which may not always be the case. I think this would be a lot more flexible if it took a String or Writable as an argument that represents the value/expression being converted, and then places that in the correct place itself.

Reply:
I am not sure I understand properly what you are suggesting here...

What do you mean with expect `.something()` to always exists?

I thought that passing a shape and adding the suffix to it in case it differs from the original shape was the most general way of achieving this.

Can you expand a little on your proposal?